#!/usr/bin/env python

PACKAGE = "bluerov2_trajectory"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       	Name		Type      	Level	Description				Default 	Min	Max

gen.add(	"traj_on",	   bool_t,	0,	             "Trajectory start flag", 		False)
gen.add("pub_on_setpoint_position",bool_t,	0,	"Publish flag on /mavros/setpoint_position/local", False)
trajectory_enum = gen.enum([ 	gen.const("pos_hold", int_t, 0, "Maintain current position"),
				gen.const("setpoint", int_t, 1, "Go to the desired waypoint"),
				gen.const("circle",   int_t, 2, "Do a circlular trajectory around a specified centre "),
                       		gen.const("custom",   int_t, 3, "Read custom trajectory from txt file "),],
                     		"An enum to set flight trajectory")
gen.add(	"reg_on",		bool_t,		0,	"Regression on flag",			False)
gen.add(	"traj_type", 		int_t, 		0, 	"Trajectory type", 			0, 	0, 	3, edit_method=trajectory_enum)
gen.add( 	"wp_x",		       double_t, 	0, 	"Waypoint X position [m]",  		0, 	-15.0, 	15.0)
gen.add( 	"wp_y",		       double_t, 	0, 	"Waypoint Y position [m]",  		0, 	-15.0, 	15.0)
gen.add( 	"wp_z",		       double_t, 	0, 	"Waypoint Z position [m]",  		0.0, 	-50.0, 	15.0)
gen.add( 	"wp_yaw",		double_t,       0, 	"Waypoint yaw angle [deg]",  	         0, 	-180, 	180)
gen.add( 	"des_radius",		double_t, 	0, 	"Radius of circle [m]",  	   	1.0, 	0.2, 	10.0)
gen.add( 	"des_velocity",		double_t, 	0, 	"Absolute velocity of Circle [m/s]",  	0.1, 	0.01, 	1.0)


exit(gen.generate(PACKAGE, "rov_trajectory", "set_trajectory"))
